core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -223,12 +289,16 @@ public void close() throws IOException {
 
         @Override
         public CloseableIterator<ScanTask> iterator() {
+          Expression partitionFilter =
```

Comment:
I am a bit worried about calling this for each manifest. Can we cache it? What do we do in other places?